### PR TITLE
Validate options in dev mode

### LINKED ISSUE #766: Validate all params passed to sentry even in dev/test
👋 
Me again, and thanks again!

It'd be great if even in dev/test mode Sentry validated all parameters/options passed. It does validate some but not all:

```
tobi@qiqi:~$ iex -S mix
iex(1)> Sentry.capture_message("hu", client: [oha: :nonsense])
:ignored
tobi@qiqi:~$ MIX_ENV=prod iex -S mix
iex(1)> Sentry.capture_message("hu", client: [oha: :nonsense])
** (NimbleOptions.ValidationError) invalid value for :client option: expected atom, got: [oha: :nonsense]
    (nimble_options 1.1.1) lib/nimble_options.ex:359: NimbleOptions.validate!/2
    (sentry 10.6.2) lib/sentry/client.ex:106: Sentry.Client.send_event/2
    iex:1: (file)
iex(1)> 

```

I suspect this behavior is because Sentry is deactivated, but it allows some bad options to slip through dev and test to production. I.e. I had somehow (don't ask me how) falsely assumed that I could pass an environment name to the client in `capture_message`'s optins: `client: [environment_name: System.get_env("ENV)]` - which made it all the way to production 😅 

So still checking that even if dsn isn't configured would be nice I think.

### PR DESCRIPTION
**Pull request by savhappy**
-validate all options passed to send_event even when in dev mode
-closes #766

### CODE REVIEW

File: lib/sentry.ex [hunk 1]
```diff
@@ -335,7 +335,7 @@ defmodule Sentry do
         Client.send_event(event, opts)
 
       !Config.dsn() ->
-        :ignored
+        Client.validate_and_ignore(opts)
```
Comment (whatyouhide) [inline lib/sentry.ex:338]: This is gonna make it harder to track the return value–I have to jump to another module. Maybe we can do

```elixir
Client.validate_options!(opts)
:ignored
```

instead?

File: lib/sentry/client.ex [hunk 2]
```diff
@@ -160,6 +160,12 @@ defmodule Sentry.Client do
     end
   end
 
+  @spec validate_and_ignore(keyword()) :: :ignored
+  def validate_and_ignore(opts) when is_list(opts) do
```
Comment (whatyouhide) [inline lib/sentry/client.ex:164]: We can also re-use this function above where we call `NimbleOptions.validate!` if we change it to return the options, like suggested in the other comment.

File: test/sentry_test.exs [hunk 3]
```diff
@@ -119,6 +119,16 @@ defmodule SentryTest do
     assert logged_count == 2
   end
 
+  test "raises error with validate_and_ignore/1 in dev mode if opts passed are invalid " do
+    put_test_config(dsn: nil, test_mode: false)
+
+    assert_raise(NimbleOptions.ValidationError, fn ->
+      Sentry.Client.validate_and_ignore(client: [bad_key: :nada])
+    end)
```
Comment (whatyouhide) [inline test/sentry_test.exs:127]: Nit:

```suggestion
    assert_raise NimbleOptions.ValidationError, fn ->
      Sentry.Client.validate_and_ignore(client: [bad_key: :nada])
    end
```

File: test/sentry_test.exs [hunk 4]
```diff
@@ -119,6 +119,16 @@ defmodule SentryTest do
...
+
+    assert :ignored == Sentry.Client.validate_and_ignore(client: :hackney)
```
Comment (whatyouhide) [inline test/sentry_test.exs:129]: Nit, but matches other tests:

```suggestion
    assert :ignored = Sentry.Client.validate_and_ignore(client: :hackney)
```

**Review by whatyouhide — COMMENTED** [review]
Left just a couple of comments but this is great.

File: lib/sentry.ex (hunk 1 above)
Comment (savhappy) [reply lib/sentry.ex:338]: Yes, good idea!

**Review by whatyouhide — APPROVED** [review]

**Pull request merged**